code/modules/mob/living/living.dm

```diff
@@ -225,7 +225,9 @@ Sorry Giacom. Please don't be mad :(
 /mob/living/proc/InCritical()
 	return (src.health < 0 && src.health > -95 && stat == UNCONSCIOUS)
 
-/mob/living/ex_act(severity, target)
+/mob/living/ex_act(severity, origin)
+	if(istype(origin, /datum/spacevine_mutation) && isvineimmune(src))
+		return
```

Comment:
Target is pretty much data argument where you can put anything and let other thigns try and react to this things accordingly
So its just ununifies system of explosion types